substrate/utils/frame/remote-externalities/src/lib.rs

```diff
@@ -402,13 +403,12 @@ where
 			})
 	}
 
-	/// Get keys with `prefix` at `block` in a parallel manner.
-	async fn rpc_get_keys_parallel(
-		&self,
-		prefix: &StorageKey,
+	/// Get keys and values with `prefix` at `block` using a stream.
+	async fn rpc_get_keys_and_values<'a>(
+		&'a self,
+		prefix: &'a StorageKey,
 		block: B::Hash,
-		parallel: usize,
-	) -> Result<Vec<StorageKey>, &'static str> {
+	) -> impl Stream<Item = Vec<KeyValue>> + 'a {
 		/// Divide the workload and return the start key of each chunks. Guaranteed to return a
 		/// non-empty list.
 		fn gen_start_keys(prefix: &StorageKey) -> Vec<StorageKey> {
```

Comment:
Why is it necessary to still divide the workload when using streams? I tried to use this and it seemed to process each chunk sequentially.

Reply:
hmmm, i am trying to build off what was done initially, it isn't sequential actually, the stream keeps receiving keys and storing them as long as they are available

Reply:
Still not sure why it's necessary to divide the workload when using the stream approach?

Reply:
Ok. How can i determine the vec of storage key using the prefix?